### PR TITLE
Remove brand-hover and text-brand-hover css classes

### PR DESCRIPTION
**Pull request by ranquild**
Related to https://github.com/metabase/metabase-private/issues/73

Migrates all places where `brand-hover` and `text-brand-hover` were used to `emotion` and `color` function. 

How to verify:
- Affected places should look like they did before (check the list below)
- Reviewing each commit is much easier

1. `NativeVariablesButton` - variables button in the native query editor
Screenshot 2023-08-02 at 15 55 05

2. `LegendHeader`

The component is used only in `Funnel` but conditionally when `actionButtons` are passed. So far I haven't found any possible case when it would be rendered. Still, I decided to migrate it as it is.

3. `QuestionEmbedWidget` - sharing button in the QB

Screenshot 2023-08-02 at 16 20 09

4. `SnippetSidebarButton` - the button to open the snippet sidebar in the native query editor

Screenshot 2023-08-02 at 16 23 25

5. `VisibilityToggler` - the toggle to expand/collapse the native query editor

Screenshot 2023-08-02 at 16 25 55

6. `SnippetRow` - a snippet item in the snippet sidebar in the native query editor

Screenshot 2023-08-02 at 16 27 43

7. `AuditSidebar` (EE) admin -> audit

Please note that index links were never used.

Screenshot 2023-08-02 at 17 14 48

8. `DataReferenceButton` - in the native query editor

Screenshot 2023-08-02 at 17 16 54

9. `AuditTableVisualization` - (EE) admin -> audit -> questions -> all questions

Screenshot 2023-08-02 at 17 19 19

10. `SnippetSidebar` - native query editor

All items appear only if you add 15 snippets or more

Screenshot 2023-08-02 at 17 24 06

11. `ViewHeader` - query run button

Screenshot 2023-08-02 at 17 26 12
Screenshot 2023-08-02 at 17 26 23

12. `QuestionNotebookButton` - the button toggle the notebook editor for GUI questions

Screenshot 2023-08-02 at 17 30 43

13. `NotebookStep` - the button to preview the query

Screenshot 2023-08-02 at 17 32 32

14. `PulseEditCards` 

`/pulse/create` - create a pulse, then go to the collection when the pulse was created and open the pulse. The UI is a bit broken as it was before. We are checking here that the attachment type options are still functional (e.g. csv...)

Screenshot 2023-08-02 at 18 26 14

15. `CodeSample` 

Enable embedding in admin settings, go to a saved question, setup embedding, go to the code samples, click on the copy button

Screenshot 2023-08-02 at 19 12 26

16. `BrowserHeader` - /browse

Screenshot 2023-08-03 at 11 23 08

17. `DashboardActions` - share, refresh, full screen, night mode buttons in the dashboard header

Also check public dashboards.

Screenshot 2023-08-03 at 12 21 33

19. `DatabaseLink - link to restore the sample db

Screenshot 2023-08-03 at 12 22 41

20. `DatabaseBrowser` - the database card in the database list in /browse

Screenshot 2023-08-03 at 12 23 49

21. `ObjectActionsSelect` - the menu trigger in the segment/metric list in admin

Screenshot 2023-08-03 at 12 25 35

22. `MetadataHeader` - the table settings icon in admin -> table metadata -> table

Screenshot 2023-08-03 at 12 27 14

23. `PulseCardPreview` - icon to remove the question from the pulse

Screenshot 2023-08-03 at 12 30 36

24. `MetadataTableColumn` - field settings icon in admin -> table metadata -> table

Screenshot 2023-08-03 at 12 31 40

25. `MetadataTableList` - icon to hide/show a table

Screenshot 2023-08-03 at 12 33 43

26. `TasksApp` - the help icon in admin -> troubleshooting -> tasks

Screenshot 2023-08-03 at 12 35 04

27. `QuestionAlertWidget` - the alert icon in the QB footer

Screenshot 2023-08-03 at 12 35 49

28. `PeopleListRow` - the icon/button to reactivate a deactivate user; admin -> people -> deactivated

Broken radio buttons aren't related to this PR

Screenshot 2023-08-03 at 12 39 44

29. `SharingPane` 

It seems that the `extensions` prop is not used at all. Still, I'm not 100% convinced, that's why I migrated it anyway.

30. `EmbedModalContent` - clickable "sharing" label at the top

Screenshot 2023-08-03 at 12 45 26

31. `BrowserCrumbs` - breadcrumbs in `/browse`

Please note that links aren't highlighted on hover both here and in `master`.

Screenshot 2023-08-03 at 12 49 27

32. `SettingsBatchForm` - the title of expandable sections in admin -> auth -> ldap

Screenshot 2023-08-03 at 12 50 19

33. `AutomaticDashboardApp` - links to the suggested x-rays

Screenshot 2023-08-03 at 12 51 22

34. `SetupCheckList` - links to items in the check list

Screenshot 2023-08-03 at 12 34 38

35. `Help` - 2 links at the top in admin -> troubleshooting -> help

Screenshot 2023-08-03 at 13 11 42

36. `ValuesYouCanReference` - popover trigger in dashboard click actions -> custom url

Screenshot 2023-08-03 at 13 14 54

37. `ArchivedItem` - icons to unarchive and delete

Screenshot 2023-08-03 at 13 16 42

38. `Calendar` - left/right icon/buttons in the calendar header

Screenshot 2023-08-03 at 13 17 53

39. `ClickMappings` - columns in "other available columns" in dashboard click actions

Screenshot 2023-08-03 at 13 20 24

40. `PasswordReveal` - reset password for a user in admin -> people without email enabled

Screenshot 2023-08-03 at 13 24 47

41. `DashboardSharingEmbedModal` - share/embed button in the dashboard header

Screenshot 2023-08-03 at 13 26 14

42. `ListItem` - used in admin -> databases

Link in the title should be highlighted

Screenshot 2023-08-03 at 13 27 39

43. `ParametersPopover` - adding new parameters to a dashboard

Screenshot 2023-08-03 at 13 34 58

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/view/QuestionNotebookButton/QuestionNotebookButton.styled.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+interface ButtonRootProps {
+  isSelected?: boolean;
+}
+
+export const ButtonRoot = styled(Button)<ButtonRootProps>`
+  color: ${props => !props.isSelected && color("text-dark")};
+
+  &:hover {
+    color: ${props => props.isSelected && color("brand")};
```
Comment (kamilmielnik) [inline frontend/src/metabase/query_builder/components/view/QuestionNotebookButton/QuestionNotebookButton.styled.tsx:13]: I think this should be inversed, see https://github.com/metabase/metabase/pull/32514/commits/58e5fc6e80246f612bec90dd8d63fc7da1afdef6

```suggestion
    color: ${props => !props.isSelected && color("brand")};
```

File: frontend/src/metabase/admin/datamodel/metadata/components/MetadataTableList/MetadataTableList.styled.tsx [hunk 2]
```diff
@@ -13,3 +15,17 @@ export const AdminListItem = styled.a<AdminListItemProps>`
...
+  color: ${color("brand")};
+  cursor: pointer;
+`;
+
+export const HideIconButton = styled(IconButtonWrapper)`
+  float: right;
+  cursor: ${props => props.disabled && "not-allowed"};
+
+  &:hover {
+    color: ${color("brand")};
```
Comment (kamilmielnik) [inline frontend/src/metabase/admin/datamodel/metadata/components/MetadataTableList/MetadataTableList.styled.tsx:29]: Previously hover effect was not applied when `disabled`: https://github.com/metabase/metabase/pull/32514/commits/39c74040aab2af33545e0415774729beac35943c#diff-941767a59c0d72015ea9fcdadf56b339bce3c91b453ce55ec0a75bf7f74be5f3L318

**Review by kamilmielnik — APPROVED** [review]
Great work  :sweat_drops:  :+1: 

I reviewed all the commits individually.
I listed the few differences I could find - not sure if they are actually issues.

File: frontend/src/metabase/query_builder/components/template_tags/SnippetSidebar/SnippetSidebar.styled.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  &:hover {
+    color: ${color("brand")};
+  }
+`;
+
+interface AddSnippetIconProps {
+  isHidden?: boolean;
+}
+
+export const AddSnippetIcon = styled(Icon)<AddSnippetIconProps>`
```
[the remark this answers is not in the document]
Comment (ranquild) [reply frontend/src/metabase/query_builder/components/template_tags/SnippetSidebar/SnippetSidebar.styled.tsx:52]: Great catch!

File: frontend/src/metabase/admin/datamodel/metadata/components/MetadataTableList/MetadataTableList.styled.tsx (hunk 2 above)
Comment (ranquild) [reply frontend/src/metabase/admin/datamodel/metadata/components/MetadataTableList/MetadataTableList.styled.tsx]: Great catch!

**Pull request merged**